### PR TITLE
Hardware Utility Streamlining

### PR DESCRIPTION
**Pull request by SiddharthKadari**
Hardware Utility class created to streamline the use of the PDP switchable channel, and the compressor feedback loop.

### CODE REVIEW

**Review by AG6GR — CHANGES_REQUESTED** [review]
HardwareUtility is a bit vague, I'd prefer this to be defined as a proper subsystem, since it's owning a hardware component. Maybe CompressorSubsystem? Likewise the PDP switchable channel is just an output component that should be owned by whichever subsystem is going to make use of it, just like any other motor controller or solenoid.

**Comment by SiddharthKadari** [comment]
> HardwareUtility is a bit vague, I'd prefer this to be defined as a proper subsystem, since it's owning a hardware component. Maybe CompressorSubsystem?

Will do. Do you think this merits an FSM system, since it would be in one state the entire time? Could I just initialize the compressor in Robot.java?

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -20,16 +21,16 @@
...
+import frc.robot.systems.CompressorSystem;
 
 /**
  * The VM is configured to automatically run this class, and to call the functions corresponding to
  * each mode, as described in the TimedRobot documentation.
  */
 public class Robot extends TimedRobot {
 	private TeleopInput input;
-	private HardwareUtility hwUtilityController;
+	private Compressor pneumaticsCompressor;
```
Comment (AG6GR) [inline src/main/java/frc/robot/Robot.java:33]: Don't need this anymore

**Review by AG6GR — APPROVED** [review]
Yeah this works, we could technically make compressor on/compressor off separate states but that might be overkill.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (SiddharthKadari) [reply src/main/java/frc/robot/Robot.java:33]: Will resolve in next PR

**Pull request merged**